### PR TITLE
feat: add encryption for connected services

### PR DESCRIPTION
**Pull request by leafty**
Add encryption to connected services.
* `client_secret` is encrypted at rest
* For each token set, `access_token` and `refresh_token` are encrypted at rest

/deploy #notest renku-ui=build/connected-services-final renku-notebooks=leafty/build-connected-services renku=release-0.52.x

### CODE REVIEW

File: components/renku_data_services/migrations/versions/788840ccdcf4_add_connected_services.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def upgrade() -> None:
         sa.Column("kind", sa.Enum("gitlab", "github", name="providerkind"), nullable=False),
         sa.Column("scope", sa.String(), nullable=False),
         sa.Column("url", sa.String(), nullable=False),
-        sa.Column("client_secret", sa.String(length=500), nullable=True),
+        sa.Column('client_secret', sa.LargeBinary(), nullable=True),
```
Comment (Panaetius) [inline components/renku_data_services/migrations/versions/788840ccdcf4_add_connected_services.py:33]: formatting error? This should be double quotes.

File: test/components/renku_data_services/connected_services/test_encryption.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+from renku_data_services.app_config import Config
+from renku_data_services.base_models import APIUser
+from renku_data_services.connected_services import apispec
+from renku_data_services.connected_services import orm as schemas
+from renku_data_services.utils.cryptography import decrypt_string
+
+
+@pytest.fixture
+def admin_api_user() -> APIUser:
```
Comment (Panaetius) [inline test/components/renku_data_services/connected_services/test_encryption.py:17]: we already have an admin_user fixture in conftest that can be reused

**Review by Panaetius — CHANGES_REQUESTED** [review]

File: components/renku_data_services/migrations/versions/788840ccdcf4_add_connected_services.py (hunk 1 above)
Comment (leafty) [reply components/renku_data_services/migrations/versions/788840ccdcf4_add_connected_services.py]: Done in 73ba30f9b838edb452ca4fb3d9a6e12bc2f42299.

File: test/components/renku_data_services/connected_services/test_encryption.py (hunk 2 above)
Comment (leafty) [reply test/components/renku_data_services/connected_services/test_encryption.py]: Done in 2a8f481361bd64a479b047af00c676f3f35d9aff.

**Review by Panaetius — APPROVED** [review]

**Pull request merged**